### ISSUE TITLE
IntEnum with NewType value type regression

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#10411]
The type of `E.A.value` in this program has changed since the last release:

```py
from typing import NewType
from enum import IntEnum

N = NewType("N", int)

class E(IntEnum):
    A = N(0)

# mypy 0.812: N
# master:     int
reveal_type(E.A.value)
```

The original inferred type `N` is more precise, so this looks like a regression. This causes false positives with mypy-protobuf.

**Comment by JukkaL** [comment]
This was introduced in #10057.
